src/tokens/wrappers/clawback/IClawback.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+interface IClawbackFunctions {
+    enum TokenType {
+        ERC20,
+        ERC721,
+        ERC1155
+    }
+
+    struct Template {
```

Comment:
`uint96` for the duration is overkill, by reducing it you can pack both booleans into the storage slots, making templates only occupy 1 single storage slot.

```
struct Template {
  bool destructionOnly;
  bool transferOpen;
  uint56 duration;
  address admin;
}
```